qualtran/bloqs/arithmetic/addition.py

```diff
@@ -155,8 +155,11 @@ def on_classical_vals(
     ) -> Dict[str, 'ClassicalValT']:
         unsigned = isinstance(self.a_dtype, (QUInt, QMontgomeryUInt))
         b_bitsize = self.b_dtype.bitsize
-        N = 2**b_bitsize if unsigned else 2 ** (b_bitsize - 1)
-        return {'a': a, 'b': int(math.fmod(a + b, N))}
+        N = 2**b_bitsize
+        if unsigned:
+            return {'a': a, 'b': int((a + b) % N)}
+        hN = N >> 1
+        return {'a': a, 'b': int(a + b + hN) % N - hN}
```

Comment:
h is half? can you provide a line comment explaining the logic here and consider using a more meaningful name than `hN`. Was the previous behavior using `math.fmod` wrong? can you add a test that would have failed with the old logic?